### ISSUE TITLE
Files with empty datasets cannot be loaded by `hsload --link`

### ISSUE THREAD

**Issue by loichuder** [HDFGroup/h5pyd#103]
When trying to load a file with a dataset set to `h5py.Empty`, I have the following error:

```
  File ".../h5pyd/h5pyd/_apps/utillib.py", line 661, in object_create_helper
    create_dataset(obj, ctx)
  File ".../h5pyd/h5pyd/_apps/utillib.py", line 334, in create_dataset
    rank = len(dset_dims)
TypeError: object of type 'NoneType' has no len()
```

May be related to #102

**Comment by loichuder** [comment]
It doesn't. This line still fails as `shape` is `None` for empty datasets:
https://github.com/HDFGroup/h5pyd/blob/jreadey-work/h5pyd/_apps/utillib.py#L347

**Comment by jreadey** [comment]
Ah ok, I see the problem.  Try the latest checkin to jreadey-work please.

**Comment by loichuder** [comment]
Works fine :ok_hand: !

**Comment by jreadey** [comment]
This is in master and released to pypi as version 0.8.4.
